### ISSUE TITLE
Major Accessibility Issues

### ISSUE THREAD

**Issue by NicktheGeek** [merbird/modal-block#2]
Hi,

I'm trying to use this block to create a popup. Everything works great when I use a mouse, but it is 100% inaccessible via keyboard. The default "button" is a `span` which relies on JS to activate. This is fine, but is not part of the tab index so keyboard users are unable to "get to" the button. I added tabindex="0" using the dev controls and was able to navigate to the "button" but I couldn't activate it.

I suggest that the "button" be given `tabindex="0"` and `role="button"` which will add it to the tab index and provide the correct role for screen readers. Additionally the script that triggers opening the modal should have an event listener for `keyUp` that triggers the modal when pressing enter. 

I suspect once I can test the modal itself, there may be some other issues. The accessible expectation is that once the modal is triggered, keyboard focus is moved to the modal, likely the title. This means the title will need tabindex="0" so it can receive focus.

Additionally while the modal is open there should be a focus trap so only the modal and it's contents are able to be in focus.

Next the expectation is that pressing the esc key will close the modal and when the modal is closed, focus goes back to the button that was used to open it.

Some additional enhancements include using aria-hidden="true" on the modal when it is closed. Change that to "aria-hidden="false" when it is open. Also add aria-controls="{id for modal}" to the button and aria-expanded="false" to the button when the modal is closed and change that to aria-expanded="true" when the modal is open. These are enhancements that improve the accessibility but are not blockers like everything before this.

**Comment by merbird** [comment]
This should be fixed I now use button instead of span
